contracts/src/registry/ETHRegistrar.sol

```diff
@@ -0,0 +1,214 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.13;
+
+import {IETHRegistrar} from "./IETHRegistrar.sol";
+import {IRegistry} from "./IRegistry.sol";
+import {IERC1155Singleton} from "./IERC1155Singleton.sol";
+import {IETHRegistry} from "./IETHRegistry.sol";
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {IPriceOracle} from "./IPriceOracle.sol";
+import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import {NameUtils} from "../utils/NameUtils.sol";
+
+contract ETHRegistrar is IETHRegistrar, AccessControl {
+    bytes32 public constant CONTROLLER_ROLE = keccak256("CONTROLLER_ROLE");
+    uint256 public constant MIN_REGISTRATION_DURATION = 28 days;
+    uint64 private constant MAX_EXPIRY = type(uint64).max;
+
+    error MaxCommitmentAgeTooLow();
+    error UnexpiredCommitmentExists(bytes32 commitment);
+    error DurationTooShort(uint64 duration);
+    error CommitmentTooNew(bytes32 commitment);
+    error CommitmentTooOld(bytes32 commitment);
+    error NameNotAvailable(string name);
+    error InsufficientValue();
+
+    IETHRegistry public immutable registry;
+    IPriceOracle public immutable prices;
+    uint256 public immutable minCommitmentAge;
+    uint256 public immutable maxCommitmentAge;
```

Comment:
We should consider making these mutable, since the contract already has an owner.

Reply:
I don't understand this comment. Unless we provide methods to change these values, what's the point of making them mutable?